pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -431,6 +431,48 @@ func TestLDROnNodeShutdown(
 
 	leftJobID, rightJobID := setupLDR(ctx, t, c, setup, ldrWorkload, mode)
 
+	findNonGatewayNode := func(info *clusterInfo, rng *rand.Rand) int {
+		for {
+			anotherNode := info.nodes.SeededRandNode(rng)[0]
+			if anotherNode != info.gatewayNodes[0] {
+				return anotherNode
+			}
+		}
+	}
+
+	var shutdownSide *clusterInfo
+	var shutdownNode int
+	if setup.rng.Intn(2) == 0 {
+		t.L().Printf("Shutting down on right side")
+		shutdownSide = setup.right
+	} else {
+		t.L().Printf("Shutting down node on left side")
+		shutdownSide = setup.left
+	}
+	if setup.rng.Intn(2) == 0 {
+		shutdownNode = findNonGatewayNode(shutdownSide, setup.rng)
+		t.L().Printf("Shutting down worker node %d", shutdownNode)
+	} else {
+		// We're assuming the gateway node is coordinating the ingestion job, as it
+		// served the create logical repl stmt.
```

Comment:
Is this implied? I thought anyone could win the race to adopt after a `CreateAdoptableJobWithTxn` call?

Reply:
good call. just added logic to control for this.